RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc

```diff
@@ -48,61 +66,157 @@ namespace {
   inline int nPixelHits(reco::Track const & tk) {
     return tk.hitPattern().numberOfValidPixelHits();
   }
-      
+  
   inline float dz(reco::Track const & trk, Point const & bestVertex) {
-      return std::abs(trk.dz(bestVertex));
+    return std::abs(trk.dz(bestVertex));
   }
   inline float dr(reco::Track const & trk, Point const & bestVertex) {
-      return std::abs(trk.dxy(bestVertex));
+    return std::abs(trk.dxy(bestVertex));
+  }
+
+  inline void dzCut_par1(reco::Track const & trk, int & nLayers, const float * par, float dzCut[]) {
+    float dzE =  trk.dzError();
+    for (int i=2; i>=0; --i) {
+      dzCut[i] = powN(par[i]*nLayers,4)*dzE;
+    }
+  }
+  inline void drCut_par1(reco::Track const & trk, int & nLayers, const float* par, float drCut[]) {
+    float drE =  trk.d0Error();
+    for (int i=2; i>=0; --i) {
+      drCut[i] = powN(par[i]*nLayers,4)*drE;
+    }
   }
+  
+  inline void dzCut_par2(reco::Track const & trk, int & nLayers, const float * par, float dzCut[]) {
+    float pt = float(trk.pt());
+    float eta = float(trk.eta());
 
+    // parametrized d0 resolution for the track pt
+    float nomd0E = sqrt(0.003*0.003+(0.001/pt)*(0.001/pt));
```

Comment:
;)
  mia

On Tue, Jan 12, 2016 at 9:06 PM, Matti Kortelainen <notifications@github.com
> wrote:

> In RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc
> <https://github.com/cms-sw/cmssw/pull/12917#discussion_r49506092>:
>
> >
> > +    // parametrized d0 resolution for the track pt
> > +    float nomd0E = sqrt(0.003*0.003+(0.001/pt)*(0.001/pt));
>
> I fully agree on the goal, but was thinking that sometimes it would be
> useful to be able to test a cut-based selector on a single iteration (e.g.
> to gain understanding on what is going on). I don't plan to push such thing
> to a release (and probably wouldn't even consider asking if it would be
> more work than adding option to configure).
>
> Thanks Mia for adding the configuration option!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12917/files#r49506092>.
>